src/cr/cube/slices.py

```diff
@@ -0,0 +1,1140 @@
+# encoding: utf-8
+
+from __future__ import division
+
+import numpy as np
+from scipy.stats.contingency import expected_freq
+
+from cr.cube.util import lazyproperty
+from cr.cube.enum import DIMENSION_TYPE as DT
+
+
+class _CatXCatSlice(object):
+    def __init__(self, dimensions, counts, base_counts):
+        self._dimensions = dimensions
+        self._counts = counts
+        self._base_counts = base_counts
+
+    @lazyproperty
+    def _row_dimension(self):
+        return self._dimensions[0]
+
+    @lazyproperty
+    def _column_dimension(self):
+        return self._dimensions[1]
+
+    @lazyproperty
+    def _row_elements(self):
+        return self._row_dimension.valid_elements
+
+    @lazyproperty
+    def _column_elements(self):
+        return self._column_dimension.valid_elements
+
+    @lazyproperty
+    def _row_generator(self):
+        return zip(self._counts, self._base_counts, self._row_elements)
+
+    @lazyproperty
+    def _column_generator(self):
+        return zip(self._counts.T, self._base_counts.T, self._column_elements)
+
+    @lazyproperty
+    def rows(self):
+        return tuple(
+            _CategoricalVector(counts, base_counts, element.label, self.table_margin)
+            for counts, base_counts, element in self._row_generator
```

Comment:
Just a quick description here: We're iterating across various rows and labels here. We need for each row the following:
- weighted count
- unweighted count
- label
- table margin
The first 3 are different for each row, while the table margin is the same. This will change for various MR cases, where the table margin changes.

We could also say that we _need_ the row margin, or the col margin for each vector here, but the fact is that the vector is smart enough (i.e. has access to enough information) that it can calculate those by itself, therefore those are implemented on the vector class.